src/main/java/br/com/deveficiente/bolaoapi/shared/validator/UniqueValidator.java

```diff
@@ -0,0 +1,61 @@
+package br.com.deveficiente.bolaoapi.shared.validator;
+
+import org.hibernate.validator.internal.engine.constraintvalidation.ConstraintValidatorContextImpl;
+import org.hibernate.validator.internal.engine.constraintvalidation.ConstraintViolationCreationContext;
+import org.springframework.stereotype.Service;
+
+import javax.persistence.EntityManager;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.stream.Stream;
+
+@Service
+public class UniqueValidator implements ConstraintValidator<Unique, String> {
+
+    private EntityManager entityManager;
+    private Class entityClass;
+
+
+    public UniqueValidator(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    @Override
+    public void initialize(Unique constraintAnnotation) {
+        entityClass = constraintAnnotation.entityClass();
+    }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        String field = getField((ConstraintValidatorContextImpl) context);
```

Comment:
*Deixamos pistas que facilitem o uso do código onde não conseguimos resolver com compilação.*
Aqui você estabelece um padrão implícito que o nome do atributo do ponto de utilização é o mesmo nome do atributo da entidade que você buscar pela comparação... Talvez possa deixar alguma pista explicando isso na sua annotation... 

Outra opção é deixar um outro parâmetro na sua annotation, com valor default vazio, abrindo a possibilidade disso ser especificado pelo usuário da api.

Reply:
Eu tinha utilziado a abordagem de buscar pelo nome do field anotado, mas como essa validação é basicamente utilizada na borda externa, caso o nome do atributo de uma classe DTO for diferente do da classe de Entidade, isso pode gerar bugs!

Tentei simplificar para o usuário, mas acabei possibilitando a geração de um bug.
Vou implementar a sugestão do parâmetro, mas vou deixa-lo obrigatório, para não dar margem da classe de DTO possuir um atributo que mapeia para o atributo da entidade com nome diferente (o que geraria o bug).